accepted/2020/workloads/workload-manifest.md

```diff
@@ -0,0 +1,430 @@
+
+# Pack Kinds
+
+The following kind of packs are permitted in manifests: targeting and runtime packs, SDK packs, library packs, template packs, and tools packs.
+
+All packs are expected to be installed into the `dotnet` folder.
+
+## Targeting and Runtime Packs
+
+The [targeting and runtime packs](https://github.com/dotnet/designs/pull/50) introduced in .NET Core 3.0 are permitted in workloads. Targeting packs contain the framework assemblies that a project compiles against at build time, and runtime packs contain the framework and runtime binaries used to execute the project. They install into `dotnet/packs/{pack-id}/{pack-version}` and are resolved at build time using `FrameworkReference` MSBuild items.
+
+Framework references are expected to be added to a project automatically by the manifest’s MSBuild targets based on the TFM. There will be a mechanism that allows the targets to omit the pack version from the framework reference. Targets may add multiple targeting or runtime pack references, for example when building a `net5.0-ios` project they could add a reference to a pack containing the `net5.0` reference assemblies and a pack containing the reference assemblies for the iOS platform bindings.
+
+## SDK Packs
+
+SDK packs are MSBuild SDK NuGet packages, but they are installed to `dotnet/packs/{pack-id}/{pack-version}`. They are expected to be automatically referenced by the targets.
+
+## Library Packs
+
+Library packs are normal NuGet packages. When a library pack is installed, the nupkg package file is placed in `dotnet/library-packs/` but are *not* extracted. This location is used as a directory feed by NuGet.
+
+If ones of these packages gets used, it will be extracted into the global packages folder. If the SDK is updated and the library pack is removed or replaced with a newer version, projects that have already been created and restored will continue to be able to use the extracted copy.
+
+This is intended to be used to pre-download NuGet packages referenced by templates so that a workload can work offline after installation. It should be used sparingly i.e. only for NuGets referenced by the core templates with default or popular options.
+
+## Template Packs
+
+Template packs are NuGet packages to be used by the dotnet templating engine. When a template pack is installed installed, the nupkg package file is placed in `dotnet/template-packs/` but is *not* extracted. Visual Studio and `dotnet new` are expected to automatically detect when templates nupkgs are installed and uninstalled to this folder and update their template hives automatically.
```

Comment:
Open: Should template packs be special cased, relative to library packs? Maybe we just need a registry of template packs.

Reply:
Library packs don't need to be partitioned into separate folders for each version band, but I'm not sure whether this is the case for template packs.

Reply:
All packs are resolved via the band manifest; the pack storage location is pretty much an implementation detail. The pack locations are split by pack kind but that isn't strictly necessary, it just makes the disk structure a bit more comprehensible.

In practice, the same pack id+version may be included in multiple bands, so the storage can be deduplicated by using the same location for all bands.

Reply:
As for extraction on demand - there are two kinds of packs and they're stored as nupkgs for different reasons.

1) Templates are stored as nupkgs is because that's what the template engine currently consumes. They could be unpacked on installation if the template engine were changed to support consuming unpacked templates in place, and I've added a note to this effect.

2) Library packs are stored as nupkgs because it's actually just a download cache in the form of the directory feed. They're handed off to NuGet and it does what it wants - currently, unpacking them into the global package cache, which is unaffected by future workload operations. This means that updating workloads will not break existing projects that reference the packs with an exact version.